client_test.go

```diff
@@ -1461,18 +1461,13 @@ func testClient_logger(t *testing.T, proto string) {
 
 // Test that we continue to consume stderr over long lines.
 func TestClient_logStderr(t *testing.T) {
-	orig := stdErrBufferSize
-	stdErrBufferSize = 32
-	defer func() {
-		stdErrBufferSize = orig
-	}()
-
 	stderr := bytes.Buffer{}
 	c := NewClient(&ClientConfig{
 		Stderr: &stderr,
 		Cmd: &exec.Cmd{
 			Path: "test",
 		},
+		LogBufferSize: 32,
```

Comment:
This tests that the existing behaviour still works, but I can't find any test coverage to confirm that JSON logs are correctly deserialised iff they are within the buffer size. It would be nice to have a test that writes one shorter and one longer JSON log, and see that the level is correctly interpreted for only the shorter one, or something along those lines.

Reply:
Thanks for your suggestion. I will update the Option name and add relevant test cases later.